.github/workflows/deploy.yml

```diff
@@ -37,9 +37,9 @@ jobs:
       
       - name: Check Release Tag Format
         run: |
-          re=[0-9]+\.[0-9]+\.[0-9]+b[0-9]+
+          re=^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(beta|alpha|rc))(\.[0-9]+)?$
           if ! [[ $RELEASE_VERSION =~ $re ]]; then
-            echo 'Tag does not match expected regex pattern for beta releases (v[0-9]+.[0-9]+.[0-9]+b[0-9]+)'
+            echo 'Tag does not match expected regex pattern for beta releases (^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(beta|alpha|rc))(\.[0-9]+)?$)'
```

Comment:
you should be able to use double quotes and `(v$re)` like i did [here](https://github.com/Bandwidth/ruby-sdk/blob/main/.github/workflows/deploy.yml#L39C68-L39C74) to avoid having the regex double typed